### PR TITLE
refactor(windows)!: only support UCRT, even for mingw

### PR DESCRIPTION
**Pull request by bfredl**
work towards #22534 .

### CODE REVIEW

**Review by dundargoc — APPROVED** [review]
Looks good. There is a `target_link_libraries(test_lib INTERFACE -municode)` in `test/functional/fixtures/CMakeLists.txt`, can that be removed as well?

File: runtime/doc/news.txt [hunk 1]
```diff
@@ -148,6 +148,10 @@ VIMSCRIPT
   treats BIN, STR and FIXSTR as separate types. Any of these is returned as a
   string if possible, or a |blob| if the value contained embedded NUL:s.
 
+BUILD SYSTEM
```
Comment (justinmk) [inline runtime/doc/news.txt:151]: The sections are in alphabetical order, this should go after "API". Could also just name it `BUILD` or maybe `CORE` or `INTERNAL`

**Comment by justinmk** [comment]
This blurb would be helpful in the commit message for each of these planned changes:

    The newer UCRT runtime has native support for UTF-8, including setting
    it as the _active codepage_ even before `main()` gets called. Whereas
    MSVCRT does not support UTF-8 and requires `wmain()`.

File: runtime/doc/news.txt (hunk 1 above)
Comment (bfredl) [reply runtime/doc/news.txt]: it is not `INTERNAL` . This note is for third-party distros (one of the main consumers of release notes)

**Review by justinmk — APPROVED** [review]

**Pull request merged**